esmf_regrid/tests/unit/experimental/io/test_round_tripping.py

```diff
@@ -150,6 +157,25 @@ def test_GridToMeshESMFRegridder_bilinear_round_trip(tmp_path):
     )
 
 
+def test_GridToMeshESMFRegridder_curvilinear_round_trip(tmp_path):
+    """Test save/load round tripping for `GridToMeshESMFRegridder`."""
+    original_rg, src = _make_grid_to_mesh_regridder(grid_dims=2)
+    filename = tmp_path / "regridder.nc"
+    save_regridder(original_rg, filename)
+    loaded_rg = load_regridder(str(filename))
+
+    assert original_rg.grid_x == loaded_rg.grid_x
+    assert original_rg.grid_y == loaded_rg.grid_y
+
+    # Demonstrate regridding still gives the same results.
+    src_data = np.arange(np.product(src.data.shape)).reshape(src.data.shape)
+    src_mask = np.zeros(src.data.shape)
+    src_mask[0, 0] = 1
+    src.data = ma.array(src_data, mask=src_mask)
+    # TODO: make this a cube comparison when mesh comparison becomes available.
+    assert np.array_equal(original_rg(src).data, loaded_rg(src).data)
```

Comment:
I think here you're just comparing data values, not whether masks matched. For that I think you need to check the equality of the masks and the data